apps/calendar/js/views/event_base.js

```diff
@@ -281,6 +281,11 @@ EventBase.prototype = {
     // always remove loading initially (to prevent worst case)
     this.element.classList.remove(this.LOADING);
 
+    // Re-run the header font fit when it comes into view.
+    // Since the header is already in the markup on load and the view is hidden
+    // the font fit calculations will be wrong initially.
+    this.header.runFontFitSoon();
```

Comment:
actually, the fact the header's text changes doesn't make this necessary, but the fact that the button's text changes makes it :)